hooks/lib/provisioning_wizard.rb

```diff
@@ -74,7 +74,7 @@ def domain
 
   def dns
     @dns ||= begin
-      line = File.read('/etc/resolv.conf').split("\n").detect { |line| line =~ /nameserver\s+.*/ && line !~ /#{ip}/ }
+      line = File.read('/etc/resolv.conf').split("\n").detect { |line| line =~ /nameserver\s+.*/ && ( line !~ /#{ip}\s*$/ || ip.nil? ) }
```

Comment:
This ensures we don't compare a nil address in case the interface doesn't have an ip address.

Also adjusted the regex so 192.168.252.10 doesn't match 192.168.252.100, 101, 102, etc. 

now it looks for the address+0 or more spaces+end of line giving a proper match.